rts/motoko-rts/src/stabilization/layout.rs

```diff
@@ -349,39 +410,48 @@ pub unsafe fn deserialize<M: Memory>(
     stable_object: StableValue,
 ) -> Value {
     let tag = stable_memory.read::<StableTag>(stable_object.to_stable_address());
-    match tag.decode() {
-        StableObjectKind::Array => {
-            StableArray::deserialize(main_memory, stable_memory, stable_object)
+    let object_kind = tag.decode();
```

Comment:
Most of the cases don't actually need the object_kind except to assert on the object_kind. 
The Array and Blob cases do need, but instead of decoding object_kind a second time, could also just be called with an appropriate Tag value instead. The additional cost is probably negligible, and the uniformity is nice,  but I guess it might be worth optimizing in future....

Reply:
Indeed, the tag is only needed for specific allocations that have multiple sub-tags (like array, blobs and boxed numbers). E.g. `StableArray::deserialize` needs to call `alloc_array` that requires the specific tag.

Reply:
Ah, I missed the fact that these additional arguments are forced by the trait. No worries.